src/hotspot/share/classfile/systemDictionaryShared.cpp

```diff
@@ -186,9 +186,19 @@ DumpTimeClassInfo* SystemDictionaryShared::get_info(InstanceKlass* k) {
 
 DumpTimeClassInfo* SystemDictionaryShared::get_info_locked(InstanceKlass* k) {
   assert_lock_strong(DumpTimeTable_lock);
-  assert(!k->is_shared(), "sanity");
+#ifdef ASSERT
+  if (DumpSharedSpaces) {
+    assert(!k->is_shared(), "sanity");
+  } else {
+    assert(DynamicDumpSharedSpaces, "sanity");
```

Comment:
I am trying to understand this block of code. Does it imply `k` can be a shared class when `DynamicDumpSharedSpaces` is set?  When can this happen? 
If `k` can be a shared class here, then the assert in the next call to `DumpTimeClassInfo::get_info` for `k` to be a non-shared class will be triggered. So there seem to be some inconsistency in the asserts.

Reply:
It turns out those changes are not needed; they are probably leftover code from an early prototype.
I also reverted the change in `SystemDictionaryShared::is_excluded_class`.